File: BookEditorDemo.Web/Scripts/ko/src/bookInfoViewModel.js
var modelBase = require('./modelBase.js');
var ko = require('knockout');

BookInfoViewModel.prototype = Object.create(modelBase.prototype)
function BookInfoViewModel(data) {
    this.id = ko.observable();
	this.title = ko.observable("");
	this.year = ko.observable("");
    this.author = ko.observable("");
	
	if (data) {
		this.id(data.Id);
		this.title(data.Title);
		this.year(data.PublicationYear);
		this.author(data.AuthorLastName + ' ' + data.AuthorFirstName);
	}

    this.description = ko.computed(function() {
        return this.title() + ' by ' + this.author()  + ', ' + this.year();
    }, this);
}


module.exports = BookInfoViewModel;

File: BookEditorDemo.Web/ViewModels/BookInfoViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace BookEditorDemo.Web.ViewModels
{
    public class BookInfoViewModel
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string AuthorFirstName { get; set; }

        public string AuthorLastName { get; set; }

        public int PublicationYear { get; set; }
    }
}